worker/deployer/nested.go

```diff
@@ -301,18 +356,18 @@ func (c *nestedContext) DeployUnit(unitName, initialPassword string) error {
 	return nil
 }
 
-func (c *nestedContext) startUnit(unitName string) error {
+func (c *nestedContext) startUnitWorkers(unitName string) error {
 	// Assumes lock is held.
 	c.logger.Infof("starting workers for %q", unitName)
-	agent, found := c.units[unitName]
-	if !found {
-		var err error
-		agent, err = c.newUnitAgent(unitName)
-		if err != nil {
-			return errors.Trace(err)
-		}
-		c.units[unitName] = agent
+	agent := c.units[unitName]
+	if agent == nil {
```

Comment:
the agent, ok pattern is preferred and more idiomatic IMO

Reply:
I think it could be ok and `agent == nil` now because we always set `c.units[unitName] = agent`.
Not sure if it's intended.